libiop/bcs/bcs_prover.tcc

```diff
@@ -78,21 +73,27 @@ void bcs_prover<FieldT, MT_hash_type>::signal_index_submissions_done()
         throw std::invalid_argument("Didn't provide prover index to BCS prover");
     }
     iop_protocol<FieldT>::signal_prover_round_done();
-    std::size_t ended_round = this->num_prover_rounds_done_-1;
-    const domain_to_oracles_map mapping = this->oracles_in_round(ended_round);
 
-    /* First, go through all the oracle messages in this round and
-       fill in the merkle tree from the index. */
-    for (auto &kv : mapping)
```

Comment:
We still have to loop through all the different domains in a round

Reply:
It is doing so in `bcs_prover::run_hashchain_for_round`